control/autoware_autonomous_emergency_braking/src/node.cpp

```diff
@@ -648,30 +650,40 @@ Path AEB::generateEgoPath(const double curr_v, const double curr_w)
   constexpr double epsilon = std::numeric_limits<double>::epsilon();
   const double & dt = imu_prediction_time_interval_;
   const double & horizon = imu_prediction_time_horizon_;
+  double path_arc_length = 0.0;
   for (double t = 0.0; t < horizon + epsilon; t += dt) {
     curr_x = curr_x + curr_v * std::cos(curr_yaw) * dt;
     curr_y = curr_y + curr_v * std::sin(curr_yaw) * dt;
     curr_yaw = curr_yaw + curr_w * dt;
     geometry_msgs::msg::Pose current_pose;
     current_pose.position = autoware::universe_utils::createPoint(curr_x, curr_y, 0.0);
     current_pose.orientation = autoware::universe_utils::createQuaternionFromYaw(curr_yaw);
-    if (autoware::universe_utils::calcDistance2d(path.back(), current_pose) < 1e-2) {
+    const double distance_between_points =
+      autoware::universe_utils::calcDistance2d(path.back(), current_pose);
```

Comment:
I think this calculation method is fine, but wouldn't using curr_v * dt provide a simpler and more accurate distance calculation?

Reply:
@tkimura4 you are totally right. I have changed the code to reflect your recommendation.

Reply:
@tkimura4 thank you, I have refactored the code to use your suggestion and also delete a lot of repeated code on that function.